src/main.c

```diff
@@ -517,19 +517,37 @@ int main(int argc, char **argv)
 			}
 			if (strcmp(long_options[option_index].name,
 			           "cookie") == 0) {
-				cli_cfg.cookie = strdup(optarg);
+				char *cookie_with_prefix;
+
+				cookie_with_prefix = strdup_with_prefix(optarg, "SVPNCOOKIE=");
+				if (cookie_with_prefix == NULL) {
+					log_error("Could not prepend \"SVPNCOOKIE=\" to the cookie.\n");
+					goto exit;
+				}
+				free(cli_cfg.cookie);
+				cli_cfg.cookie = cookie_with_prefix;
```

Comment:
I would rather keep this PR focused on prepending SVPNCOOKIE, without adding checks:
````c
				free(cli_cfg.cookie);
				cli_cfg.cookie = strdup_with_prefix(optarg, "SVPNCOOKIE=");
````
I'm not saying these checks are bad. Just keep them for a different PR.

Reply:
OK, I understand we can add `free(cookie)` in #1041, right? Perhaps in this case you can add it, because all the surrounding code is changed. Besides, for consistency, I think you should add `free(cookie);` and remove `free(cli_cfg.cookie);`.

Reply:
Right, I removed too much stuff. I kept `free(cli_cfg.cookie);` simply because it was there already, but I can remove it to make the code consistent.

Reply:
Keep it if it was already there. Looks good.